app/views/site/get_new_annotations.js.erb

```diff
@@ -1,3 +1,10 @@
+// dump out annotation dropdown values to see if we have a match for the existing selected annotation
+var annotationData = <%= raw @cluster_annotations.values.map {|coll| coll.map(&:last)}.flatten %>;
```

Comment:
As part of the long trek towards SCP UI 2.0, I'd like to gradually start separating javascript code from ruby dynamic templates as much as possible.   So this line can be split into 
```
var selectedAnnotationElement = <%= .. %>
var selectedAnnotation = selectedAnnotationElement.val()
```

This will slowly make it clearer exactly what data we need from the server so we can build the appropriate APIs

(For bonus credit, the above line could be rewritten to put the ruby computation of the value.map on a controller, so that the template can just say `var annotationData = <%= @flattenedAnnotationValues %>
`

Reply:
Makes sense - I can do that.